ui/ui-frontend/projects/archive-search/src/app/archive/archive-preview/archive-unit-information-tab/archive-unit-information-tab.component.ts

```diff
@@ -35,43 +35,105 @@
 import { ArchiveService } from '../../archive.service';
 
 @Component({
   selector: 'app-archive-unit-information-tab',
   templateUrl: './archive-unit-information-tab.component.html',
   styleUrls: ['./archive-unit-information-tab.component.css'],
 })
-export class ArchiveUnitInformationTabComponent implements OnInit, OnChanges {
+export class ArchiveUnitInformationTabComponent implements OnInit, OnChanges, OnDestroy {
   @Input() archiveUnit: Unit;
-  @Input() accessContractId: string;
   @Input() tenantIdentifier: number;
 
   @Output() showNormalPanel = new EventEmitter<any>();
 
   uaPath$: Observable<{ fullPath: string; resumePath: string }>;
   fullPath = false;
   hasDownloadDocumentRole = false;
+  private accessContractAllowDownloadData: boolean = null;
 
-  constructor(private archiveService: ArchiveService) {}
+  private accessContract: AccessContract;
+  private subscription: Subscription;
+
+  constructor(
+    private archiveService: ArchiveService,
+    private accessContractService: AccessContractService,
+  ) {}
 
   ngOnInit() {
+    this.getAccessContract();
     this.checkDownloadPermissions();
     this.uaPath$ = this.archiveService.buildArchiveUnitPath(this.archiveUnit);
   }
 
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+
   ngOnChanges(changes: SimpleChanges): void {
     if (changes.archiveUnit?.currentValue['#id']) {
       this.uaPath$ = this.archiveService.buildArchiveUnitPath(this.archiveUnit);
+      this.accessContractAllowDownloadData = null;
     }
     this.fullPath = false;
   }
 
-  onDownloadObjectFromUnit(archiveUnit: Unit) {
-    return this.archiveService.downloadObjectFromUnit(archiveUnit['#id']);
+  private getAccessContract() {
+    this.subscription = this.accessContractService.currentAccessContract$.subscribe({
+      next: (accessContract: AccessContract) => {
+        this.accessContract = accessContract;
+        this.accessContractAllowDownload();
+      },
+      error: (e) => console.error(e),
+    });
+  }
+
+  accessContractAllowDownload(): boolean {
+    if (this.accessContractAllowDownloadData) {
+      return this.accessContractAllowDownloadData;
+    }
+    if (!this.accessContract || !this.archiveUnit.objectGroup || !this.archiveUnit.objectGroup.versionsWithQualifiers) {
+      return false;
+    }
+    this.accessContractAllowDownloadData = this.accessContractAllowDownloadFor(this.archiveUnit);
+    return this.accessContractAllowDownloadData;
+  }
+
+  private accessContractAllowDownloadFor(unit: Unit): boolean {
+    if (this.accessContract.everyDataObjectVersion) {
+      return true;
+    }
+    if (!this.accessContract.dataObjectVersion) {
+      return false;
+    }
+    for (let versionWithQualifier of unit.objectGroup.versionsWithQualifiers) {
+      if (this.accessContract.dataObjectVersion.includes(versionWithQualifier.qualifier)) {
+        return true;
+      }
+    }
```

Comment:
Ne retourne aucune valeur si on sort de la boucle.
Peut être simplifié :
```suggestion
    return unit.objectGroup.versionsWithQualifiers.some(versionWithQualifier => this.accessContract.dataObjectVersion.includes(versionWithQualifier.qualifier));
```